validator/main.py

```diff
@@ -46,3 +46,42 @@ def validate(self, value: Any, metadata: Dict = {}) -> ValidationResult:
                 fix_value="{The programmtic fix if applicable, otherwise remove this kwarg.}",
             )
         return PassResult()
+
+    def _inference_local(self, model_input: Any) -> Any:
+        """
+        Runs a machine learning pipeline on some input on the local
+        machine. This function should receive the expected input to the
+        ML model, and output the results from the ml model.
+        """
+        raise NotImplementedError
+
+    def _inference_remote(self, model_input: Any) -> Any:
+        """
+        Runs a machine learning pipeline on some input on a remote
+        machine. This function should receive the expected input to the
+        ML model, and output the results from the ml model.
+
+        """
+        raise NotImplementedError
+
+    def _inference(self, model_input: Any) -> Any:
```

Comment:
I think we need to rework how this and the base [Validator](https://github.com/guardrails-ai/guardrails/blob/main/guardrails/validator_base.py) class interact.  Maybe by specifying any methods we want/need the validator author to override as abstract.  I would prefer to keep the higher level orchestration methods like this one in house vs requiring the author to define it.